app/Http/Controllers/QuizAPIController.php

```diff
@@ -51,4 +49,26 @@ function addQuiz(Request $request)
             ], 500);
         }
     }
+
+    function singleQuiz(int $id)
+    {
+        try {
+            $quiz = Quiz::with(['questions' => ['answers'],])->find($id);
+        } catch (Exception $e) {
+            return response()->json([
+                'message' => 'Internal server error'
+            ], 500);
+        }
+
+        if (!is_null($quiz)) {
```

Comment:
`null` is falsy in PHP so you could also write this as:

```suggestion
        if ($quiz) {
```